examples/kalman_filter.py

```diff
@@ -0,0 +1,374 @@
+            theta = gs.array([theta])
+        theta = gs.transpose(theta)
+        return Localization.group.rotations.matrix_from_rotation_vector(theta)
+
+    @staticmethod
+    def regularize_angle(theta):
+        """Bring back angle theta in ]-pi, pi]."""
+        if gs.ndim(gs.array(theta)) <= 1:
+            theta = gs.array([theta])
+        return Localization.group.rotations.log_from_identity(theta.T)
+
+    def __init__(self):
+        self.dim = self.group.dim
+        self.dim_noise = 3
+        self.dim_obs = 2
+
+    def adjoint_map(self, state):
+        """Construct the tangent map associated to Ad_X : g |-> XgX^-1."""
+        theta, x, y = state
+        tangent_base = gs.array([[0, -1],
+                                 [1, 0]])
+        ad = gs.eye(3)
+        ad[1:, 1:] = self.rotation_matrix(theta)
+        ad[1:, 0] = -tangent_base.dot([x, y])
+
+        return ad
+
+    def propagate(self, state, sensor_input):
+        """Propagate state with constant velocity motion model on SE(2)."""
+        dt, linear_speed, angular_speed = self.split_input(sensor_input)
+        theta, x, y = state
+        x, y = state[1:] + dt * self.rotation_matrix(theta).dot(
+            linear_speed)
+        theta = theta + dt * angular_speed
+        theta = self.regularize_angle(theta)
+        return gs.concatenate((theta, [x, y]))
+
+    def propagation_jacobian(self, state, sensor_input):
+        """Construct the jacobian associated to the input.
+
+        Since the propagation writes f(x) = x*u, and the error is modeled on
+        the Lie algebra, the jacobian is Ad_{u^{-1}}.
+        """
+        dt, linear_speed, angular_speed = self.split_input(sensor_input)
+        input_vector_form = dt * gs.hstack((angular_speed, linear_speed))
+        input_inv = self.group.inverse(input_vector_form)
+
+        return self.adjoint_map(input_inv)
+
+    def noise_jacobian(self, state, sensor_input):
+        """Construct the jacobian associated to the process noise.
+
+        The noise being considered multiplicative, it is simply the identity
+        scaled by the time stamp.
+        """
+        dt, _, _ = self.split_input(sensor_input)
+        return gs.sqrt(dt) * gs.eye(self.dim_noise)
+
+    def observation_jacobian(self, state, observation):
+        """Construct the jacobian associated to the innovation."""
+        return gs.eye(2, 3, 1)
+
+    def get_measurement_noise_cov(self, state, observation_cov):
+        """Construct the measurement covariance for the innovation."""
+        theta, _, _ = state
+        rot = self.rotation_matrix(theta)
+        return rot.T.dot(observation_cov).dot(rot)
+
+    @staticmethod
+    def observation_model(state):
+        """Model used to obtain the measurements."""
+        return state[1:]
+
+    def innovation(self, state, observation):
+        """Discrepancy between the measurement and its expected value.
+
+        Here the linear error observation - expected is brought back to
```

Comment:
Should this be a todo? Hard to understand the comment here.

Reply:
It is not a todo, more an explanation of why it is different than the linear case and the standard EKF, following the reference given at the beginning. Should I give a more direct reference, then (e.g., equation number) ? Or simply go back to a more simple approach ?

Reply:
I see. Might be the formulation of the sentence. Maybe state clearly that 'the 'expected' vector is brought back to the Lie algebra. (Also can we find something more accurate than 'brought'? It doesn't sound like a proper mathematical term).